astropy/coordinates/transformations.py

```diff
@@ -347,23 +347,34 @@ def get_transform(self, fromsys, tosys):
             self._composite_cache[fttuple] = comptrans
         return self._composite_cache[fttuple]
 
-    def lookup_name(self, name):
+    def lookup_name(self, name, raise_exception=False):
         """
         Tries to locate the coordinate class with the provided alias.
 
         Parameters
         ----------
         name : str
             The alias to look up.
-
+        raise_exception : Bool (default value is False)
+            False : doesn't raise exception if 'name' isn't found. (returns None)
+            True : raises exception if 'name' isn't found.
         Returns
         -------
         coordcls
+        if raise_exception == False:
             The coordinate class corresponding to the ``name`` or `None` if
             no such class exists.
+        if raise_exception == True:
+            The coordinate class corresponding to the ``name`` or raises Exception
+            if no such class exists.
         """
-
-        return self._cached_names.get(name, None)
+        if not raise_exception:
+            return self._cached_names.get(name, None)
+        else:
+            if name in self._cached_names:
+                return self._cached_names[name]
+            else:
+                raise NameError("name not found")
```

Comment:
Not sure if a `NameError` is appropriate here, `NameError`s are for accesses to undefined variables. Maybe just use `return self._cached_names` and just let the `KeyError` "happen".